xcode/Subconscious/Shared/Services/DatabaseService.swift

```diff
@@ -257,6 +257,50 @@ final class DatabaseService {
         return results.first?.col(0)?.toInt()
     }
     
+    func listEntriesForSlashlinks(slashlinks: [Slashlink]) throws -> [EntryStub] {
+        guard self.state == .ready else {
+            throw DatabaseServiceError.notReady
+        }
+
+        let parameters =
+            slashlinks
+            .flatMap { s in
+                // TODO: this is almost certainly the wrong way to do this
+                [
+                    SQLite3Database.Value.text(s.toPublicMemoAddress().description),
+                    SQLite3Database.Value.text(s.toLocalMemoAddress().description)
+                ]
```

Comment:
As the TODO says, this doesn't seem very clever. I imagine this will be affected by https://github.com/subconsciousnetwork/subconscious/pull/561

Reply:
Yep, calling it in a loop also felt dumb to me but it's probably the lesser of two evils if performance isn't going to be an issue.